lib/middlewares/src/metering.rs

```diff
@@ -75,29 +111,45 @@ impl<F: Fn(&Operator) -> u64 + Copy + Clone + Send + Sync + 'static> ModuleMiddl
             remaining_points_index: self.remaining_points_index.lock().unwrap().expect(
                 "Metering::generate_function_middleware: Remaining points index not set up.",
             ),
+            error_code_index: self.error_code_index.lock().unwrap().expect(
+                "Metering::generate_function_middleware: Error code index not set up.",
+            ),
             accumulated_cost: 0,
         })
     }
 
     /// Transforms a `ModuleInfo` struct in-place. This is called before application on functions begins.
     fn transform_module_info(&self, module_info: &mut ModuleInfo) {
         let mut remaining_points_index = self.remaining_points_index.lock().unwrap();
-        if remaining_points_index.is_some() {
+        let mut error_code_index = self.error_code_index.lock().unwrap();
+
+        if remaining_points_index.is_some() || error_code_index.is_some() {
             panic!("Metering::transform_module_info: Attempting to use a `Metering` middleware from multiple modules.");
         }
 
         // Append a global for remaining points and initialize it.
-        let global_index = module_info
+        let remaining_points_global_index = module_info
             .globals
             .push(GlobalType::new(Type::I64, Mutability::Var));
-        *remaining_points_index = Some(global_index.clone());
+        *remaining_points_index = Some(remaining_points_global_index.clone());
+
+        let error_code_global_index = module_info
+            .globals
+            .push(GlobalType::new(Type::I32, Mutability::Var));
+        *error_code_index = Some(error_code_global_index.clone());
+
         module_info
             .global_initializers
             .push(GlobalInit::I64Const(self.initial_limit as i64));
 
         module_info.exports.insert(
             "remaining_points".to_string(),
-            ExportIndex::Global(global_index),
+            ExportIndex::Global(remaining_points_global_index),
+        );
+
+        module_info.exports.insert(
+            "error_code".to_string(),
```

Comment:
Small nit, if these global export names can conflict with user global variables, lets use something less likely to conflict, like prefixing them with `wasmer_metering_`